File: solutions/javascript/stack/05-size/stack.js
class Stack {
  constructor() {
    this._items = [];
  }

  push(value) {
    this._items.push(value);
  }

  pop() {
    if (this.isEmpty()) {
      return undefined;
    }
    return this._items.pop();
  }

  peek() {
    if (this.isEmpty()) {
      return undefined;
    }
    return this._items[this._items.length - 1];
  }

  size() {
    return this._items.length;
  }

  isEmpty() {
    return this._items.length === 0;
  }
}

module.exports = Stack;

File: solutions/python/stack/04_peek/stack.py
class Stack:
    def __init__(self) -> None:
        self._items = []

    def push(self, value) -> None:
        self._items.append(value)

    def pop(self):
        if not self._items:
            return None
        return self._items.pop()

    def peek(self):
        if not self._items:
            return None
        return self._items[-1]

